### PR TITLE
Telescience Restoration - Modularized

### PR DESCRIPTION
**Pull request by BlueNothing**
Restores the functionality of Telescience to all stations by adding the telepad and control console to the techweb as a node after Bluespace Travel and Miniaturized Bluespace. Now modularized!
/🆑

[why]: Telescience is a famous and well-loved part of the SS13 experience that has been replicated only in hollow fashion in recent days with devices like the quantum pad (which still has a useful niche if you want A->B->C->A transportation, or A->B->A transportation without a second person involved), the bluespace launchpad (which mostly everyone treats as useless, because its range means that it rarely gets you anywhere interesting unless you're already somewhere at least almost as interesting, but which does have a much simpler control scheme), or the traitor briefcase launchpad (which is still somewhat useful, if only due to being portable and letting you yoink things from another room (pairing well with x-ray or thermal vision)). A proper restoration would likely be no shortage of amusement to many.

### CODE REVIEW

**Comment by BlueNothing** [comment]
This version seems quieter.

**Comment by niceping** [comment]
So, "passive bluespace shielding" when? Would be nice as: >flag for areas to be shielded, including entire centcom and secure areas like ai sat or engine >item that protecting 3x3 area around it from telepad

I just trying to prevent salt and telesci getting nerfed to shit again

**Comment by BlueNothing** [comment]
Centcom Z-level is already shielded.

**Comment by BlueNothing** [comment]
As for secure areas, I'm not sure how important/necessary it is to shield them. Most of the 'interesting' places you'd want to teleport to are probably considered secure, especially when you consider that wormhole guns are a prerequisite for telescience, and wormhole guns are enough to enter the armory with relative ease.

**Comment by BlueNothing** [comment]
I do like the idea of an item that can produce shielded zones, though.

**Comment by BlueNothing** [comment]
Besides, even if you rush telescience, you're going to need...
Industrial Engineering, Data Theory, Bluespace 1, Bluespace 2, Bluespace Travel 1, Hi-Eff Parts, and Bluespace 3 (which is expensive at the moment), before you can even pick up the required tech. You get roughly 1.5 techs a minute for the first 6, then Bluespace 3 and Bluespace Travel 2 are basically another 5 techs worth, so you've got about ten minutes to secure anything valuable (like, say, just moving it out of the armory into another room, or having your borg move you somewhere else on the AI sat as AI). Telecomms, gravity generator... those are the two main spots that are remain at issue, if any. AI Upload? By the time you can do that, RnD could've just researched the tech to relaw the AI from wherever. 

Basically, telescience is very powerful, but it's not something you can just pull out of a hat. There's enough time for people to make minor changes to the locations of the important stuff by the time it could be stolen that way.

**Comment by MrWankman** [comment]
Why is CentCom level shielded? Personally I feel that since CentCom has an entire station itself that it should be able to be interacted with to some degree by someone dedicated enough to get there before shuttle landing.

**Comment by BlueNothing** [comment]
It is interacted with. Cargo gets their stuff from them, a sufficiently
intrepid murderboner summons an ERT or death squad from them, etc. A
*dedicated* adventurer can still get there... but I see no reason to allow
telescience to be the cheesy way to turn Centcomm into a loot run.

**Comment by MrWankman** [comment]
Oh, so that level is accessible via travel then?

**Comment by BlueNothing** [comment]
Not by telescience, but I think you could hijack an ERT ship? Not by normal
means of travel, though. Oh, and the nuke-op ship also parks on that
Z-level. Do you want people trying to rush nuke-ops before they even launch?

**Comment by MrWankman** [comment]
Oh that's right it's the admin level also. Still I find amusement in people being able to somehow get to these kinds of places that are off limits or just hard to get to in this game. lol

**Comment by yskefunky** [comment]
Not 100% on this because I remember the shittery of just nuking a malf ai within a minute, but I don't mind giving it a try. +1

**Comment by Pyko1** [comment]
needs to be fixed

**Comment by BlueNothing** [comment]
Fixed how?

On Nov 12, 2018 10:55 AM, "Pyko" <notifications@github.com> wrote:

needs to be fixed

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/HippieStation/HippieStation/pull/9433#issuecomment-437932206>,
or mute the thread

**Comment by Pyko1** [comment]
Conflicting files
code/modules/shuttle/supply.dm
hippiestation.dme
hippiestation/code/modules/research/techweb/all_nodes.dm

**Comment by BlueNothing** [comment]
Oh, thanks for telling me there were conflicts, I hadn't known. I'll have
to fix them later.

**Comment by GuyonBroadway** [comment]
**CONFLICTS!!**

**Comment by BlueNothing** [comment]
Produced what should be a fixed version here: https://github.com/HippieStation/HippieStation/pull/9853
